app/controllers/incomeSources/add/CheckBusinessDetailsController.scala

```diff
@@ -16,62 +16,232 @@
+            } else {
+              Logger("application").error(
+                s"[CheckBusinessDetailsController][handleRequest] - Error: ${ex.getMessage}")
+              itvcErrorHandler.showInternalServerError()
+            }
+        }
+      }
+    }
+  }
+
+  def submit(): Action[AnyContent] = (checkSessionTimeout andThen authenticate andThen retrieveNino
+    andThen retrieveIncomeSources andThen retrieveBtaNavBar).async {
+    implicit user =>
+      getDetails(user).toOption match {
+        case Some(viewModel: CheckBusinessDetailsViewModel) =>
+          businessDetailsService.createBusinessDetails(viewModel) map {
+          case Left(ex) => Logger("application").error(
+            s"[CheckBusinessDetailsController][handleRequest] - Unable to create income source: ${ex.getMessage}")
+            itvcErrorHandler.showInternalServerError()
+
+          case Right(IncomeSource(id)) =>
+            user.session.data - "addBusinessName" - "addBusinessStartDate" - "addBusinessTrade" - "addBusinessAddressLine1" - "addBusinessPostalCode" - "addBusinessAccountingMethod"
+            Redirect(controllers.incomeSources.add.routes.CheckBusinessDetailsController.changeBusinessReportingMethod().url + s"?IncomeSourceID=$id")
+        }
+        case None => Logger("application").error(
+          s"[CheckBusinessDetailsController][submit] - Error: Unable to build view model on submit")
+          Future.successful(itvcErrorHandler.showInternalServerError())
+      }
+  }
+
+  def submitAgent: Action[AnyContent] = Authenticated.async {
+    implicit request =>
+      implicit user =>
+        getMtdItUserWithIncomeSources(incomeSourceDetailsService).flatMap {
+          implicit mtdItUser =>
+            getDetails(mtdItUser).toOption match {
+              case Some(viewModel: CheckBusinessDetailsViewModel) =>
+                businessDetailsService.createBusinessDetails(viewModel) map {
+                  case Left(ex) => Logger("application").error(
+                    s"[CheckBusinessDetailsController][handleRequest] - Unable to create income source: ${ex.getMessage}")
+                    itvcErrorHandler.showInternalServerError()
+
+                  case Right(_) =>
+                    mtdItUser.session.data - "addBusinessName" - "addBusinessStartDate" - "addBusinessTrade" - "addBusinessAddressLine1" - "addBusinessPostalCode" - "addBusinessAccountingMethod"
+                    Redirect(controllers.incomeSources.add.routes.CheckBusinessDetailsController.changeBusinessReportingMethodAgent())
+                }
+              case None => Logger("application").error(
+                s"[CheckBusinessDetailsController][submit] - Error: Unable to build view model on submit")
+                Future.successful(itvcErrorHandler.showInternalServerError())
+            }
+        }
+  }
+
+
+  def changeBusinessName(): Action[AnyContent] = Action {
+    Ok("Change Business Name WIP")
```

Comment:
we need implementation to where were are going to redirect for each of these methods;
I would assume we have most of these controllers already created, so relevant public methods should be created otherwise create one with no implementation;

Reply:
As discussed will leave these as they are new unimplemented change pages, can be updated once work is there